### PR TITLE
oddjob: init at 0.34.7

### PR DESCRIPTION
**Pull request by s1341**
###### Description of changes

The odd job dameon for dbus. https://pagure.io/oddjob

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/admin/oddjobd.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  config = mkIf cfg.enable {
+    systemd.packages = [ pkgs.oddjob ];
+
+    systemd.services.oddjobd = {
+      wantedBy = ["multi-user.target"];
+      after = ["network.target"];
+      description = "DBUS Odd-job Daemon";
+      enable = true;
+      documentation = ["man:oddjobd(8)" "man:oddjobd.conf(5)"];
+      #serviceConfig = {
```
Comment (SohamG) [inline nixos/modules/services/admin/oddjobd.nix:29]: Are you sure this needs to be commented out?

Comment (s1341) [reply nixos/modules/services/admin/oddjobd.nix:29]: No. But without it I think we try to create the service twice...

Comment (SohamG) [reply nixos/modules/services/admin/oddjobd.nix:29]: Can you supress creation of the default .service file? Maybe nix is installing it when the package builds and then the option is installing it again with the mkIf

**Comment by SohamG** [comment]
Bump for visibility. Waiting on this for realmd

File: nixos/modules/services/admin/oddjobd.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  pkg = pkgs.oddjob;
```
Comment (IvarWithoutBones) [inline nixos/modules/services/admin/oddjobd.nix:6]: A configurable `package` option would be nice.

File: pkgs/os-specific/linux/oddjob/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  ];
+
+  postConfigure = ''
+    substituteInPlace src/oddjobd.c --replace "globals.selinux_enabled" "FALSE"
+  '';
+
+  meta = with lib; {
+    description = "Odd Job Daemon";
+    homepage = "https://pagure.io/oddjob";
+    changelog = "https://pagure.io/oddjob/blob/master/f/ChangeLog";
```
Comment (IvarWithoutBones) [inline pkgs/os-specific/linux/oddjob/default.nix:36]: Could this be fixed to the branch of the version packaged instead?

File: nixos/modules/services/admin/oddjobd.nix (hunk 1 above)
Comment (IvarWithoutBones) [reply nixos/modules/services/admin/oddjobd.nix:29]: Any reason we don't just use the one upstream provides?

File: nixos/modules/services/admin/oddjobd.nix (hunk 2 above)
Comment (s1341) [reply nixos/modules/services/admin/oddjobd.nix:6]: What other package would be used? Or do you mean for overrides?

File: pkgs/os-specific/linux/oddjob/default.nix (hunk 3 above)
Comment (s1341) [reply pkgs/os-specific/linux/oddjob/default.nix:36]: The changelog link?

File: nixos/modules/services/admin/oddjobd.nix (hunk 2 above)
Comment (IvarWithoutBones) [reply nixos/modules/services/admin/oddjobd.nix:6]: Just so people can easily use overrides. Most other modules have an `options.*.package` option.

File: pkgs/os-specific/linux/oddjob/default.nix (hunk 3 above)
Comment (IvarWithoutBones) [reply pkgs/os-specific/linux/oddjob/default.nix:36]: Yes.

**Comment by SohamG** [comment]
@s1341 Please can I get write access to this branch? (assuming you are no longer wanting to work on this)

**Comment by IvarWithoutBones** [comment]
@SohamG you can always branch off of their fork and create your own PR. In that case you should keep the existing commits to retain credits to the original author of this PR, and can commit your own changes seperately.

**Comment by s1341** [comment]
Apologies. I don't have time to get to this at the moment. @SohamG if you still want write perms, let me know.

**Comment by SohamG** [comment]
@s1341 Yes please can I have write perms?

**Pull request closed without merging**